### ISSUE TITLE
The `-H:+DumpTargetInfo` argument causes build failure on Windows when working directory and Graal are on different drives

### ISSUE THREAD

**Issue by collinpeters** [oracle/graal#3286]
**Describe the issue**
The `-H:+DumpTargetInfo` causes builds on Windows to fail with `IllegalArgumentException: 'other' has different root`.

**Steps to reproduce the issue**
Please include both build steps as well as run steps
1. On Windows install GraalVM on C:\
2. Build your native-image from a working directory other than C:\

**Describe GraalVM and your environment:**
 - GraalVM version 20.2.0
 - JDK major version: 11
 - OS: Windows
 - Architecture: AMD64

**More details**
When working on building static images for Linux I added the `-H:+DumpTargetInfo` to the build params to get target info. This seemed to cause Windows builds to fail. Looking at `VMFeature.java` (from the stacktrace) I see this code:
```
        if (SubstrateOptions.DumpTargetInfo.getValue()) {
            System.out.println("# Static libraries:");
            Path current = Paths.get(".").toAbsolutePath();
            nativeLibraries.getStaticLibraries().stream().map(current::relativize).map(Path::toString).forEach(x -> System.out.println("#   " + x));
            System.out.println("# Other libraries: " + String.join(",", nativeLibraries.getLibraries()));
        }
```
The `relativize` method will cause failure as it cannot 'traverse' different roots (e.g. C: and D:) on Windows.

```
[nexus-iq-cli:6692]     analysis: 182,533.25 ms,  2.96 GB
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:600)
	at java.base/java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1006)
	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:480)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:349)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:508)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:114)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:537)
Caused by: java.lang.IllegalArgumentException: 'other' has different root
	at java.base/sun.nio.fs.WindowsPath.relativize(WindowsPath.java:404)
	at java.base/sun.nio.fs.WindowsPath.relativize(WindowsPath.java:42)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at com.oracle.svm.hosted.VMFeature.afterAnalysis(VMFeature.java:92)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$9(NativeImageGenerator.java:753)
	at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:70)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:753)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:555)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:468)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Error: Image build request failed with exit status 1
com.oracle.svm.driver.NativeImage$NativeImageError: Image build request failed with exit status 1
	at com.oracle.svm.driver.NativeImage.showError(NativeImage.java:1558)
	at com.oracle.svm.driver.NativeImage.build(NativeImage.java:1308)
	at com.oracle.svm.driver.NativeImage.performBuild(NativeImage.java:1269)
	at com.oracle.svm.driver.NativeImage.main(NativeImage.java:1228)
	at com.oracle.svm.driver.NativeImage$JDK9Plus.main(NativeImage.java:1740)
```

**Comment by oubidar-Abderrahim** [comment]
I see you're using an old version of GraalVM, did you try this using one of the latest versions? (20.3.1.2 or 21.0.0.2)

**Comment by oubidar-Abderrahim** [comment]
Internally tracked in GR-30312

**Comment by collinpeters** [comment]
> I see you're using an old version of GraalVM, did you try this using one of the latest versions? (20.3.1.2 or 21.0.0.2)

I have not. I see that the call to `relativize` is still present in the 21.0.0.2 `VMFeature` so I think it will still be there.

**Comment by oubidar-Abderrahim** [comment]
This should be fixed by f55915ac2638559b52d4b369c470f98028f66d72 
If not please reopen.
